internal/controller/prefectserver_controller.go

```diff
@@ -113,271 +123,161 @@ func (r *PrefectServerReconciler) Reconcile(ctx context.Context, req ctrl.Reques
-			"Job", desiredMigrationJob.Name, server.Name,
-		)
-		condition = conditions.AlreadyExists(objName, errorMessage)
-		return &ctrl.Result{}, errors.NewBadRequest(errorMessage)
-
 	case !isMigrationJobFinished(foundMigrationJob):
 		log.Info("Waiting on active migration Job to complete", "name", foundMigrationJob.Name)
-		condition = conditions.AlreadyExists(objName, fmt.Sprintf("migration Job %s is still active", foundMigrationJob.Name))
+		meta.SetStatusCondition(&server.Status.Conditions, conditions.AlreadyExists(objName, fmt.Sprintf("migration Job %s is still active", foundMigrationJob.Name)))
 
 		// We'll requeue after 20 seconds to check on the migration Job's status
 		return &ctrl.Result{Requeue: true, RequeueAfter: 20 * time.Second}, nil
 
 	default:
 		if !meta.IsStatusConditionTrue(server.Status.Conditions, "MigrationJobReconciled") {
-			condition = conditions.Updated(objName)
+			meta.SetStatusCondition(&server.Status.Conditions, conditions.Updated(objName))
 		}
 	}
 
 	return nil, err
 }
 
 func (r *PrefectServerReconciler) reconcileDeployment(ctx context.Context, server *prefectiov1.PrefectServer, desiredDeployment appsv1.Deployment, log logr.Logger) (*ctrl.Result, error) {
-	var condition metav1.Condition
-	var err error
-	defer func() {
-		if updateErr := r.updateCondition(ctx, server, condition); updateErr != nil {
-			err = utilerrors.NewAggregate([]error{updateErr, err})
-		}
-	}()
-
 	objName := constants.Deployment
 
-	serverNamespacedName := types.NamespacedName{
-		Namespace: server.Namespace,
-		Name:      server.Name,
+	deploy := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      server.Name,
+			Namespace: server.Namespace,
+		},
 	}
 
-	foundDeployment := &appsv1.Deployment{}
-	err = r.Get(ctx, serverNamespacedName, foundDeployment)
-	if errors.IsNotFound(err) {
-		log.Info("Creating Deployment", "name", desiredDeployment.Name)
-		if err = r.Create(ctx, &desiredDeployment); err != nil {
-			condition = conditions.NotCreated(objName, err)
-
-			return &ctrl.Result{}, err
+	result, err := controllerutil.CreateOrUpdate(ctx, r.Client, deploy, func() error {
+		if err := ctrl.SetControllerReference(server, deploy, r.Scheme); err != nil {
+			return err
 		}
 
-		server.Status.Version = prefectiov1.VersionFromImage(desiredDeployment.Spec.Template.Spec.Containers[0].Image)
-
-		condition = conditions.Created(objName)
-	} else if err != nil {
-		condition = conditions.UnknownError(objName, err)
-
-		return &ctrl.Result{}, err
-	} else if !metav1.IsControlledBy(foundDeployment, server) {
-		errorMessage := fmt.Sprintf(
-			"%s %s already exists and is not controlled by PrefectServer %s",
-			"Deployment", desiredDeployment.Name, server.Name,
-		)
+		return mergo.Merge(deploy, desiredDeployment, mergo.WithOverride)
+	})
 
-		condition = conditions.AlreadyExists(objName, errorMessage)
+	log.Info("CreateOrUpdate", "object", objName, "name", server.Name, "result", result)
```

Comment:
Bah, I think I may have lost some logging in the other operator, thanks for the reminder